application/src/main/java/org/opentripplanner/updater/trip/siri/SiriRealTimeTripUpdateAdapter.java

```diff
@@ -334,4 +428,32 @@ private boolean markScheduledTripAsDeleted(Trip trip, final LocalDate serviceDat
 
     return success;
   }
+
+  /**
+   * Types of SIRI update messages.
+   */
+  private enum SiriUpdateType {
```

Comment:
What do you think of Henrik's idea of using an algebraic data type?

```java
sealed interface SiriUpdate

record TripUpdate(
  VehicleJourney vehicleJourney
  ...
) implements SiriUpdate

record ReplacementDeparture implementes SiriUpdate
```

Reply:
Sounds great. Henrik has a draft PR for this, I would rather keep it simple in this PR.